src/utils/manager.py

```diff
@@ -178,10 +194,11 @@ def mount(
         Error: Raised if the mount operation fails.
     """
     # Convert `endpoint` to the `mount.nfs` format if `endpoint` is an NFS URL.
-    if _nfs_url_check.match(endpoint):
-        endpoint, port = _translate(endpoint)
+    if tl := _translate(endpoint):
+        endpoint, port = tl
     else:
-        port = None
+        _logger.error(f"Cannot translate invalid endpoint url {endpoint}")
+        raise Error(f"Cannot translate invalid endpoint url {endpoint}")
```

Comment:
Still deciding if throwing an error is better than GIGO. Would like to hear your opinion, @NucciTheBoss.

Reply:
IMO catching & reporting errors early and often is my preferred approach. 

If a bad NFS URL does make it into the client, I'd prefer that we'd catch that asap, exit and report to the admin in `juju status` that we have a bad NFS URL rather than try to mount and fail. Helps to narrow down what the specific issue is so that the resolution is more clear. E.g. "hey, we have a bad URL!" rather than "hey it could be a variety of reasons such as your network configuration is bad, your NFS server isn't actually up, or you forgot to enable NFS mounts in LXD."